Support a few more Cython system packages

orlitzky opened the pull request:
The usual stuff, for:

- cypari2
- cysignals
- fpylll
- memory_allocator
- pplpy
- primecountpy

### Dependencies

- https://github.com/sagemath/sage/pull/36482

tobiasdiez committed:
Loosen version requirement on fpylll and align its conda version

orlitzky committed:
build/pkgs/memory_allocator: add standard python spkg-configure.m4

orlitzky committed:
build/pkgs/cysignals: add standard python spkg-configure.m4

orlitzky committed:
build/pkgs/fpylll: add standard python spkg-configure.m4

orlitzky committed:
build/pkgs/pplpy: add standard python spkg-configure.m4

orlitzky committed:
build/pkgs/primecountpy: add standard python spkg-configure.m4

orlitzky committed:
build/pkgs/cypari: add standard python spkg-configure.m4

orlitzky committed:
build/pkgs/memory_allocator: add Gentoo package information

orlitzky committed:
build/pkgs/cysignals: add Gentoo package information

orlitzky committed:
build/pkgs/primecountpy: add Gentoo package information

orlitzky committed:
build/pkgs/pplpy: add Gentoo package information

orlitzky committed:
build/pkgs/fpylll: add Gentoo package information

orlitzky committed:
build/pkgs/cypari: add Gentoo package information

dimpase reviewed (approved):
looks good.

please put the correct tags up.

mkoeppe commented on build/pkgs/primecountpy/spkg-configure.m4:
```diff
@@ -0,0 +1,5 @@
+SAGE_SPKG_CONFIGURE([primecountpy], [
+  SAGE_SPKG_DEPCHECK([cysignals primecount], [
```
Not necessarily for this PR, but a possible next refinement:
`primecount` is only used via `primecountpy`; so instead of listing it in `SAGE_SPKG_DEPCHECK`, we could mark `primecount` as "not required" when `primecountpy` is available -- like we do for `virtualenv` as the dependency of only `tox`.

dimpase replied:
how do we do this? where is this "not required" thing should be?

orlitzky replied:
You can see an example in e.g. `filelock/spkg-configure.m4`. In theory we could do this for any package but it requires hard-coding the entire dependency tree in m4 and even I am not in a rush to do that. This is a relatively easy case though.

orlitzky committed:
build/pkgs/memory_allocator: rename the Gentoo package

Someone renamed this with a hyphen instead of an underscore a few
minutes after I added it.

mkoeppe reviewed (approved):
I guess this is fine, as the feature is marked experimental.
